source/config/configSpec.py

```diff
@@ -187,6 +187,7 @@
 [UIA]
 	enabled = boolean(default=true)
 	useInMSWordWhenAvailable = boolean(default=false)
+	consoleUIA = boolean(default=false)

```

Comment:
> This setting is only temporary and "use when available" is misleading (as that isn't what it does). The final PR will change the setting to something like:
> `winConsoleImplementation= option("auto", "legacy", "UIA", default="auto")`

Mmmm, honestly I don't see why `use when available` is misleading. In fact, the current approach allows every windows 7 or windows 8.x user to enable this option without any warning, likely getting major issues in command consoles on these versions. Even when this option is hidden in the advanced category, I really prefer a firmer constraint.

If I understand you correctly:

* Auto is supposed to be UIA if the Windows 10 version supports it properly
* UIA always uses UIA, regardless what Windows version is in use
* Legacy always uses legacy

In this case, auto can be explained as `use UIA when available and working OK`. This is similar to how the useInMSWord boolean works. Again, we really should avoid using UIA for command consoles in version of Windows where UIA is most definitely not supported for them.

Please also note that changing the option from boolean to option would require a config spec change and a config sscheme upgrade. I'd really prefer if we could reach consensus about something for the config spec that could be persistent from the start.

Reply:
Ah, I probably should've clarified a few things:

* Windows 10 has had UIA console support since version 1709, but it was incomplete in early builds (see #7497). The final PR will set a minimum good Windows version for UIA, which likely won't be 1709 (even though 1709 has *limited* support). That is to say, we won't turn on UIA console support automatically until a Windows version where we know it works completely, but we should give users the option of forcibly enabling it if our default is too aggressive.
* I think that @feerrenrut and @michaelDCurran were planning on incrementally merging UIA console support into `master` during development (similar to UIA word support). We obviously don't want to enable it by default during development (so we don't break production systems), but once it's ready we want to enable it for all users (unless they explicitly turn it off). TO accomplish this, I proposed the provisional setting during development (and then adding the auto-detection and new setting once it's final), but maybe there's a better way?

Reply:
Checkbox checked is mapped to UIA. Checkbox unchecked is mapped to auto. The label of the checkbox now has the word "force" in it.